### PR TITLE
OCPQE-18085: IPI install OCP to a shared VPC in GCP using minimal permissions

### PR DESCRIPTION
**Pull request by jianli-wei**
- add the workflow `cucushift-installer-rehearse-gcp-ipi-xpn-minimal-permission` and the provision chain
- add the step `gcp-provision-xpn-min-perm-passthrough-serviceaccount` to enable the IAM service account for the testing
- update the step `ipi-install-install` & `ipi-deprovision-deprovision` to use the above service account
- add periodic tests in 4.14-4.16 multi nightly: 
4.14 `gcp-ipi-xpn-minimal-permission-amd-f28-destructive` & `gcp-ipi-xpn-minimal-permission-arm-f28`
4.15 `gcp-ipi-xpn-minimal-permission-amd-f28` & `gcp-ipi-xpn-minimal-permission-arm-f28-destructive`
4.16 `gcp-ipi-xpn-minimal-permission-amd-f28-destructive` & `gcp-ipi-xpn-minimal-permission-arm-f28`
_FYI 4.15 "destructive" with "arm" instead is for better deviation of the cron days._

### CODE REVIEW

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-gcp-ipi-xpn-minimal-permission-f28

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-gcp-ipi-xpn-minimal-permission-f28

File: ci-operator/step-registry/gcp/provision/xpn-minpt-serviceaccount/gcp-provision-xpn-minpt-serviceaccount-commands.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "$(date -u --rfc-3339=seconds) - Enabling the IAM service account of minimal permissions for deploying OCP cluster into GCP shared VPC..."
+
+# For now using the same service account as the user-tags testing, 
+# but it is possible to use another service account in future. 
+cp "${CLUSTER_PROFILE_DIR}/user_tags_sa.json" "${SHARED_DIR}/xpn_minpt_sa.json"
```
Comment (jianlinliu) [inline ci-operator/step-registry/gcp/provision/xpn-minpt-serviceaccount/gcp-provision-xpn-minpt-serviceaccount-commands.sh:11]: `minpt` means `minimal permissions` ?

Comment (jianli-wei) [reply ci-operator/step-registry/gcp/provision/xpn-minpt-serviceaccount/gcp-provision-xpn-minpt-serviceaccount-commands.sh:11]: Sorry, "minpt" means mimimal permissions for credentials mode being Passthrough. Shall I update it? And your suggestion? Thanks!

Comment (jianlinliu) [reply ci-operator/step-registry/gcp/provision/xpn-minpt-serviceaccount/gcp-provision-xpn-minpt-serviceaccount-commands.sh:11]: my suggestion is `xpn_min_perm_passthrough.json`

**Comment by jianlinliu** [comment]
The same suggestion as https://github.com/openshift/release/pull/46874/files#r1432613187, I guess we can place the job in common ci.

**Comment by jianlinliu** [comment]
@jianli-wei  https://github.com/openshift/release/pull/47011 get merged, so I guess need to rebase and add the jobs to 4.16 too.

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/xpn/minimal-permission/provision/cucushift-installer-rehearse-gcp-ipi-xpn-minimal-permission-provision-chain.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    - ref: gcp-provision-xpn-min-perm-passthrough-serviceaccount
+    - chain: ipi-install
+    - ref: enable-qe-catalogsource
+    - chain: cucushift-installer-check
+    - ref: cucushift-installer-check-gcp-credentials-mode
+  env:
+  - name: CREDENTIALS_MODE
+    default: "Passthrough"
+    documentation: CCO mode.
+  - name: COMPUTE_NODES_TAGS
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/xpn/minimal-permission/provision/cucushift-installer-rehearse-gcp-ipi-xpn-minimal-permission-provision-chain.yaml:17]: We do not need this setting, right?

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-minimal-permission-arm-f28-destructive

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/xpn/minimal-permission/provision/cucushift-installer-rehearse-gcp-ipi-xpn-minimal-permission-provision-chain.yaml (hunk 2 above)
Comment (jianli-wei) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/xpn/minimal-permission/provision/cucushift-installer-rehearse-gcp-ipi-xpn-minimal-permission-provision-chain.yaml:17]: The service account in use has minimal permissions, and it is lack of the permissions to create/delete firewall-rules in the shared VPC. In this case, the installer won't create firewall-rules in the shared VPC, and the cluster machines are expected to use the pre-configured firewall-rules, which have targetTags configured. For example, the control-plane machines has the network tag "preserved-ipi-xpn-control-plane", and the access targeting to these machines would be allowed per the pre-configured firewall-rules. 

_FYI_
```
$ gcloud --project openshift-qe-shared-vpc compute firewall-rules list --format="table(creationTimestamp.date('%Y-%m-%d %H:%M:%S'):sort=1,name,targetTags)" --filter="network=installer-shared-vpc AND name~preserved-ipi-xpn" 2> /dev/null
CREATION_TIMESTAMP   NAME                                TARGET_TAGS
2022-10-13 02:39:55  preserved-ipi-xpn-api               ['preserved-ipi-xpn-control-plane', 'preserved-ipi-xpn-compute']
2022-10-13 02:46:23  preserved-ipi-xpn-etcd              ['preserved-ipi-xpn-control-plane']
2022-10-13 02:46:29  preserved-ipi-xpn-control-plane     ['preserved-ipi-xpn-control-plane']
2022-10-13 02:46:35  preserved-ipi-xpn-health-checks     ['preserved-ipi-xpn-control-plane', 'preserved-ipi-xpn-compute']
2022-10-13 02:46:44  preserved-ipi-xpn-internal-cluster  ['preserved-ipi-xpn-control-plane', 'preserved-ipi-xpn-compute']
2022-10-13 02:46:50  preserved-ipi-xpn-internal-network  ['preserved-ipi-xpn-control-plane', 'preserved-ipi-xpn-compute']
2022-10-17 20:45:23  preserved-ipi-xpn-bastion-access    ['preserved-ipi-xpn-bastion']
$

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/xpn/minimal-permission/provision/cucushift-installer-rehearse-gcp-ipi-xpn-minimal-permission-provision-chain.yaml:17]: Got it. thanks for the explaination.

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Comment by jianli-wei** [comment]
FYI The pre test succeeds.
Test name | Detail | Failure Analysis
-- | -- | --
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-minimal-permission-arm-f28-destructive | [link](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/46967/rehearse-46967-periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-minimal-permission-arm-f28-destructive/1737742546127294464) | INFO[2023-12-21T08:45:22Z] Step phase pre succeeded after 51m20s.

**Pull request merged**